src/docs/runtime-existential-containers.md

```diff
@@ -0,0 +1,69 @@
+
+| Index | Contents       | Size in Machine Words |
+|-------|----------------|-----------------------|
+|   0   | Payload        | 3                     |
+|   3   | Metadata       | 1                     |
+|   4   | Witness Tables | variable: 0 or more   |
+
+The start of the box includes 3 machine words which serve as a payload for the existential container. If the payload is 3 machine words or less, it will be copied into the existential container. If the payload is larger than 3 machine words, a copy will be allocated from the heap and the pointer to that copy will live in the payload.
+
+After the payload is the type metadata of the entity in the payload.
+
+Finally, there is an inline list of pointers to protocol witness tables. This list can be 0 elements in length as in the case of the special `Any` protocol or it can be n elements in length as in the case of a protocol composition type (`P1 & P2 & ... Pn`). In the case of an existential container with more than 1 protocol witness table, they are ordered alphabetically by the protocol name (note, this is from the source code - Apple has not documented this).
+
+Because of the memory issues, existential containers need to be handled with attention paid to reference counting, but this can be managed by avoiding exposing the existential containers to the public API. This is straight-forward since in order to present protocols back and forth to each language will require dual proxies. This is detailed [here](binding-protocols.md).
+
+To start, we want to be able to operate on existential containers in the abstract. Therefore all existential containers should implement a common interface:
+```csharp
+public interface IExistentialContainer {
+    // These 3 elements hold the payload of the existential type
+    IntPtr Payload0 { get; set; }
+    IntPtr Payload1 { get; set; }
+    IntPtr Payload2 { get; set; }
+    // This is the type metadata of the object in the payload. The type metadata will determine
+    // whether or not the payload contains the actual object or a heap-allocated copy. This is done by
+    // looking at the value witness table, if it's a value type, which has a field for the type's size
+    TypeMetadata ObjectMetadata { get; set; }
+    // an indexer into each of the value witness tables
+    NativeHandle this [int index] { get; set; } // this[] and Count could just as easily be IEnumerable<NativeHandle>
+    // the number of value witness tables
+    int Count { get; }
+    // the size of this in MACHINE WORDS not bytes
+    int SizeOf { get; }
```

Comment:
this = existential container? Also why we need the machine words size?

Reply:
misread my own code is why